genie-web/src/main/java/com/netflix/genie/web/spring/autoconfigure/tasks/leader/LeaderAutoConfiguration.java

```diff
@@ -241,4 +245,32 @@ public AgentJobCleanupTask agentJobCleanupTask(
             registry
         );
     }
+
+    /**
+     * Create a {@link ClusterLeaderService} bean if one is not already defined and if {@link LeaderInitiator} is
+     * available.
+     *
+     * @param leaderInitiator the Spring Zookeeper/Curator based leader election component
+     * @return a {@link ClusterLeaderService}
+     */
+    @Bean
+    @ConditionalOnBean(LeaderInitiator.class)
+    @ConditionalOnMissingBean(ClusterLeaderService.class)
+    public ClusterLeaderService clusterLeaderService(final LeaderInitiator leaderInitiator) {
```

Comment:
Seems like it might be good to have one based on local leader that forces it not to be the leader?